arith/src/bn254.rs

```diff
@@ -163,6 +166,44 @@ impl FieldSerde for Fr {
     }
 }
 
+impl FieldSerde for G1Affine {
+    const SERIALIZED_SIZE: usize = 32;
```

Comment:
This is taken from #191, I am a bit unsure if my G1/G2-Affine serialized size is correct...

but for a larger perspective - do we actually need the serialize size anymore?  Is this constant used anywhere else in the codebase, or we can just eliminate it altogether?

Reply:
32 is correct. 

https://github.com/PolyhedraZK/Expander/blob/main/arith/mersenne31/src/m31.rs#L42

this is used by the deserializer. 

I am actually not sure about the following:

```
 fn deserialize_from<R: Read>(reader: R) -> FieldSerdeResult<Self> {
        let bytes: Vec<u8> = Vec::deserialize_from(reader)?;
        let mut encoding = <Self as GroupEncoding>::Repr::default();
        encoding.as_mut().copy_from_slice(bytes.as_ref());
        match G1Affine::from_bytes(&encoding).into_option() {
            Some(a) => Ok(a),
            None => Err(FieldSerdeError::DeserializeError),
        }
    }
```

I have not used this type of deserialization before. it seems that you read all bytes from R. maybe it is sufficient to check the length of `encoding` here?

Reply:
fair - see https://github.com/PolyhedraZK/Expander/pull/200/commits/48b0df2dafa16dfac526d28057ad0b0827ca8444

I checked against the bytes read from the reader, should match the defined `SERIALIZED_SIZE`.